examples/feature_selection/plot_rfe_digits.py

```diff
@@ -3,34 +3,198 @@
-rfe.fit(X, y)
-ranking = rfe.ranking_.reshape(digits.images[0].shape)
+svc.fit(X_train, y_train)
+y_pred = svc.predict(X_test)
+accuracy_all_features = accuracy_score(y_test, y_pred)
+
+print(f"Accuracy using all {X_train.shape[1]} features: {accuracy_all_features:.4f}")
+
+# %%
+# Feature Selection with RFE
+# --------------------------
+#
+# Now, we'll employ :class:`~sklearn.feature_selection.RFE` to select a subset of
+# the most discriminative features. The goal is to determine if a reduced set of
+# important features can either maintain or even improve the classifier's performance.
+
+# %%
+from sklearn.feature_selection import RFE
+from sklearn.model_selection import GridSearchCV
+from sklearn.pipeline import Pipeline
+
+# Define the parameters for the grid search
+param_grid = {"rfe__n_features_to_select": [1, 5, 10, 20, 30, 40, 50, 64]}
+
+# Create a pipeline with feature selection followed by SVM
+pipe = Pipeline(
+    [
+        ("rfe", RFE(estimator=SVC(kernel="linear", C=1))),
+        ("svc", SVC(kernel="linear", C=1)),
+    ]
+)
+
+# Create the grid search object
+grid_search = GridSearchCV(pipe, param_grid, cv=5, scoring="accuracy", n_jobs=-1)
+
+# Fit to the data and get the best estimator
+grid_search.fit(X_train, y_train)
+best_pipeline = grid_search.best_estimator_
 
-# Plot pixel ranking
+# Extract the optimal number of features from the best estimator
+optimal_num_features = best_pipeline.named_steps["rfe"].n_features_
+
+print(f"Optimal number of features: {optimal_num_features}")
+
+# %%
+# Evaluating SVM on Selected Features
+# -----------------------------------
+#
+# With the top features selected by :class:`~sklearn.feature_selection.RFE`, let's
+# train a new :class:`~sklearn.svm.SVC` and assess its performance. The idea is to
+# observe if there's any significant change in accuracy, ideally aiming for improvement.
+
+# %%
+best_pipeline.fit(X_train, y_train)
+y_pred_rfe = best_pipeline.predict(X_test)
+
+# Get accuracy of model using selected features
+accuracy_selected_features = accuracy_score(y_test, y_pred_rfe)
+
+# get num selected features
+selected_features = best_pipeline.named_steps["rfe"].support_
+num_features_to_select = selected_features.sum()
+
+print(
+    f"Accuracy using {num_features_to_select} selected features:"
+    f" {accuracy_selected_features:.4f}"
+)
+
+# %%
+# Visualizing Feature Importance after RFE
+# ----------------------------------------
+#
+# :class:`~sklearn.feature_selection.RFE` provides a ranking of the features based on
+# their importance. We can visualize this ranking to gain insights into which pixels
+# (or features) are deemed most significant by :class:`~sklearn.feature_selection.RFE`
+# in the digit classification task.
+
+# %%
+ranking = best_pipeline.named_steps["rfe"].ranking_.reshape(digits.images[0].shape)
 plt.matshow(ranking, cmap=plt.cm.Blues)
 plt.colorbar()
 plt.title("Ranking of pixels with RFE")
 plt.show()
```

Comment:
According to the documentation `ranking_` is an array where the most important features are assigned rank 1 and the higher ranking the less important. Notice that in the current version of the example you have shades of blue going from 1 to 64 (we have 8 x 8 pixels) whereas your code uses a model which already truncated the feature space to keep only the 5 x 5 most relevant pixels and degenerating the rest to a value of `1`. I am afraid this was not the spirit of the example.